service/storage_service.cc

```diff
@@ -1116,11 +1110,24 @@ void storage_service::handle_state_normal(inet_address endpoint) {
     if (_gossiper.uses_host_id(endpoint)) {
         auto host_id = _gossiper.get_host_id(endpoint);
         auto existing = _token_metadata.get_endpoint_for_host_id(host_id);
+        auto replace_addr = db().local().get_replace_address();
         if (db().local().is_replacing() &&
```

Comment:
Because existing code handles replacing node differently. I thought it is easier to handle them in different code path. We probably can combine most of the logic. I will work on it.